### DISCUSSION TITLE
Skip Ahead

### DISCUSSION THREAD

**Discussion by giscus** [aduros/wasm4#146] [Games]
# Skip Ahead

Build retro games using WebAssembly for a fantasy console

https://wasm4.org/play/skipahead/?giscus=9cf67e59d607d31e3a4701e7yrcVs6nz3JzaBgNm6lzak6JnLtu9cwgUCa05R04W125jPVls8yK7gY9CTz7dwhPydxxE6AHdLSt2j5hCQfFLNmxMRCUCCXjBsJ1TqtPxNJfzWBktxZGDPDV%2FrIc%3D

**Comment by exoticorn** [comment]
X: Jump
Z: Restart when dead

This is a port of my [256byte TIC-80 game](http://tic80.com/play?cart=1735) "Skip Ahead" to hand optimized WebAssembly. The resulting module is exactly 512 bytes in size.
I did the port as a code example while developing a simple programming language to make writing WebAssembly by hand more comfortable while retaining full control over the resulting code. The source code to this game can be found [here](https://github.com/exoticorn/curlywas/blob/master/examples/wasm4/skipahead.cwa), but be warned that a) the language is still very much WIP and b) the code is optimized for size, not readability.

**Comment by ghost** [comment]
Tight

**Comment by misdake** [comment]
Wow, I played a lot of Skyroads when I was a kid. Cool game in such a small size budget.

**Reply by ScottFromDerby** [reply]
SKYROADS! I forgot that existed, what an amazing game

**Comment by Bouda-Lycaon** [comment]
Can you port this to a boot rom?
